dgraph/cmd/alpha/http_test.go

```diff
@@ -196,6 +196,37 @@ func queryWithTs(queryText, contentType, debug string, ts uint64) (string, uint6
 	return string(output), startTs, err
 }
 
+func queryWithTsForCost(queryText, contentType, debug string, ts uint64) (string,
```

Comment:
rather than making this a new function, you should probably pass options to queryWithTs(). Or if you don't want to do update all the callers.

```golang
type validationOpts struct {
  ResponseCallback func() 
}

func _queryWithTs(..., opts *validationOpts) {
...
context := map[string]interface{}
  if opt.ResponseCallback {
    err := opt.ResponseCallback(context, response)
  }

  return ..., context, error
}

func Test...() {
  ...
  queryWithTs(..., &callbackOptions{ResponseCallback: func() { // put cost into context }})
}
```

Reply:
I've made it to return resp instead of passing the validation function.